pydantic/_internal/_dataclasses.py

```diff
@@ -123,7 +124,7 @@ def __init__(__dataclass_self__: PydanticDataclass, *args: Any, **kwargs: Any) -
     return True
 
 
-def is_builtin_dataclass(_cls: type[Any]) -> bool:
+def is_builtin_dataclass(_cls: type[Any]) -> TypeGuard[type[StandardDataclass]]:
```

Comment:
Nice! But does this tell us that it is a pydantic datacalss? E.g. `is_builtin_dataclass(bool)`. It's probably never called like that though.

Reply:
It tells us specifically that it is _not_ a pydantic dataclass. But if it returns true you still know the thing passed is a StandardDataclass. So I think it's okay to annotate like this.

Reply:
Seems to me like it's the other way around: it tells us specifically that it _is_ a standard library dataclass, but it does _not_ tell us if it is a pydantic dataclass

Reply:
I'll also note that as of now the function is not actually used anywhere. So maybe we should just delete it anyway. (It's a hold-over from v1)

Reply:
`        and not hasattr(_cls, '__pydantic_validator__')`
can this be the case for pydantic dataclasses?